modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,276 @@
+            return
+
+        if "*" in extensions or "**" in extensions:
+            extensions = sorted(ext for ext in self.bot.extensions.keys() if ext not in blacklisted)
+
+        await ctx.send(self.batch_manage(Action.UNLOAD, *extensions))
+
+    @extensions_group.command(name="reload", aliases=("r",))
+    async def reload_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Reload extensions given their fully qualified or unqualified names.
+
+        If an extension fails to be reloaded, it will be rolled-back to the prior working state.
+
+        If '*' is given as the name, all currently loaded extensions will be reloaded.
+        If '**' is given as the name, all extensions, including unloaded ones, will be reloaded.
+        """
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        if "**" in extensions:
+            extensions = self.all_extensions.keys()
+        elif "*" in extensions:
+            extensions = [*extensions, *sorted(self.bot.extensions.keys())]
+            extensions.remove("*")
+
+        await ctx.send(self.batch_manage(Action.RELOAD, *extensions))
+
+    @extensions_group.command(name="list", aliases=("all", "ls"))
+    async def list_extensions(self, ctx: Context) -> None:
+        """
+        Get a list of all extensions, including their loaded status.
+
+        Grey indicates that the extension is unloaded.
+        Green indicates that the extension is currently loaded.
+        """
+        embed = Embed(colour=Colour.blurple())
+        embed.set_author(
+            name=f"{self.type.capitalize()} List",
+        )
+
+        lines = []
+        categories = self.group_extension_statuses()
+        for category, extensions in sorted(categories.items()):
+            # Treat each category as a single line by concatenating everything.
+            # This ensures the paginator will not cut off a page in the middle of a category.
+            category = category.replace("_", " ").title()
+            extensions = "\n".join(sorted(extensions))
+            lines.append(f"**{category}**\n{extensions}\n")
+
+        log.debug(f"{ctx.author} requested a list of all {self.type}s. " "Returning a paginated list.")
+
+        # since we currently don't have a paginator.
+        await ctx.send("".join(lines) or f"( There are no {self.type}s installed. )")
+
+    def group_extension_statuses(self) -> t.Mapping[str, str]:
+        """Return a mapping of extension names and statuses to their categories."""
+        categories = defaultdict(list)
+
+        for ext in self.all_extensions:
+            if ext in self.bot.extensions:
+                status = ":green_circle:"
+            else:
+                status = ":red_circle:"
+
+            root, name = ext.rsplit(".", 1)
+
+            category = " - ".join(root.split("."))
+
+            categories[category].append(f"{status}  {name}")
```

Comment:
```suggestion
            root, name = ext.rsplit(".", 1)
            category = " - ".join(root.split("."))
            categories[category].append(f"{status}  {name}")
```

Reply:
Due to discord markdown, this is needed.

Reply:
Huh? I just removed the empty lines in between to group them into a single section as they are one.